src/DynamoCoreWpf/Views/Preview/InfoBubbleView.xaml.cs

```diff
@@ -1023,39 +1024,79 @@ private void ScrollViewer_PreviewMouseWheel(object sender, MouseWheelEventArgs e
             {
-                if (infoBubbleDataPacket.Style != InfoBubbleViewModel.Style.Info ||
-                    ViewModel.DismissedMessages.Contains(infoBubbleDataPacket))
-                {
-                    continue;
-                }
+                ViewModel.DismissedMessages.RemoveAt(i);
+            }
+        }
 
+        /// <summary>
+        /// Repopulates the node's collection of dismissed messages.
+        /// </summary>
+        /// <param name="style"></param>
+        private void RecreateDismissedMessagesOfStyle(InfoBubbleViewModel.Style style)
+        {
+            foreach (InfoBubbleViewModel.InfoBubbleDataPacket infoBubbleDataPacket in ViewModel.NodeMessages)
+            {
+                if (infoBubbleDataPacket.Style != style) continue;
                 ViewModel.DismissedMessages.Add(infoBubbleDataPacket);
             }
+        }
+
+        /// <summary>
+        /// Combines the clearing and repopulation functions of a node's dismissed messages collection.
+        /// </summary>
+        private void RefreshDismissedMessages(InfoBubbleViewModel.Style style)
+        {
+            ClearDismissedMessagesOfStyle(style);
+            RecreateDismissedMessagesOfStyle(style);
+        }
+
+        /// <summary>
+        /// Click event handler for when the user clicks 'Dismiss All' on a node's info messages.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void DismissAllInfoButton_Click(object sender, RoutedEventArgs e)
+        {
+            // Clearing and recreating all existing info-level messages from the dismissed collection.
+            RefreshDismissedMessages(InfoBubbleViewModel.Style.Info);
 
             ViewModel.RefreshNodeInformationalStateDisplay();
         }
 
+        /// <summary>
+        /// Click event handler for when the user clicks 'Dismiss All' on a node's warnings.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
         private void DismissAllWarningsButton_Click(object sender, RoutedEventArgs e)
         {
-            foreach (InfoBubbleDataPacket infoBubbleDataPacket in ViewModel.NodeMessages)
-            {
-                if (infoBubbleDataPacket.Style != InfoBubbleViewModel.Style.Warning &&
-                    infoBubbleDataPacket.Style != InfoBubbleViewModel.Style.WarningCondensed ||
-                    ViewModel.DismissedMessages.Contains(infoBubbleDataPacket))
-                {
-                    continue;
-                }
-
-                ViewModel.DismissedMessages.Add(infoBubbleDataPacket);
-            }
+            // Clearing and recreating all existing warning-level messages from the dismissed collection.
+            RefreshDismissedMessages(InfoBubbleViewModel.Style.Warning);
+            RefreshDismissedMessages(InfoBubbleViewModel.Style.WarningCondensed);
```

Comment:
hmm I thought this style is obsoleted, do we still need to refresh it?

Reply:
The Condensed styles? They're catered for so as to not break any part of the existing API, but yes they're effectively no longer necessary.